bson/src/main/org/bson/codecs/pojo/ConventionAnnotationImpl.java

```diff
@@ -85,6 +87,15 @@ private void processPropertyAnnotations(final ClassModelBuilder<?> classModelBui
                 classModelBuilder.idPropertyName(propertyModelBuilder.getName());
             } else if (annotation instanceof BsonIgnore) {
                 propertyModelBuilder.readName(null);
+            } else if (annotation instanceof BsonRepresentation) {
```

Comment:
I decided to only check for `BsonRepresentation` in the read annotations, because I felt that there was no reason users would want to only put the tag on the setter method. This annotation feels kind of similar to `BsonId` to me which is also not checked for in the write annotations. The properties that are checked for in both do different things depending on the context (i.e. change the `readName` or the `writeName`). Because `BsonRepresentation` should do the same thing in both contexts, I thought it made sense to only allow it (and check for it) in one place.